CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+cmake_minimum_required(VERSION 3.13)
+project(concurrentqueue VERSION 0.1 LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 11)
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+
+enable_testing()
+
+add_library(concurrentqueue INTERFACE)
+
+# -Wall -pedantic-errors -Wpedantic -Wconversion -fno-elide-constructors
+target_compile_options(concurrentqueue INTERFACE -Wall -pedantic-errors -Wpedantic -Wconversion -fno-elide-constructors)
+
+target_include_directories(concurrentqueue INTERFACE
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
+)
+target_link_libraries(concurrentqueue INTERFACE Threads::Threads rt)
+
+include(CMakePackageConfigHelpers)
```

Comment:
line 21 to end of file is all about install and generate of .cmake for header only lib